cdk/log_storage/apps_script_code/log_upload_to_s3.gs

```diff
@@ -0,0 +1,77 @@
+
+// This function must be run as a Google Apps Script (currently being run from a student's account - potentially could move to Makerspace's google account)
+
+// Must first install S3 Library from: https://engetc.com/projects/amazon-s3-api-binding-for-google-apps-script/
+
+// Can be automatically executed on a schedule using "Triggers" in Apps Script
+
+
+// This function gathers data from a Google sheet and puts a csv in a specified S3 bucket
+function uploadToS3() {
+
+  // Get the data from the Google Sheet - Replace SheetID and SheetName (Not the name of the sheets document, but the actual sheet tab name - located at the bottom of Google Sheets document) 
+  var sheetId = '<YOUR_SHEET_ID>';
+  var sheetName = '<YOUR_SHEET_NAME>';
+  var sheet = SpreadsheetApp.openById(sheetId).getSheetByName(sheetName);
+  
+  fileName = sheet.getName() + ".csv";
+
+  // convert all available sheet data to csv format
+  var csvFile = convertRangeToCsvFile_(fileName, sheet);
+  
+  //Third party library named S3 is used to send data to S3.
+
+  // replace with your own IAM User's access key and secret access key
+  // This IAM role should preferably only have access to your one target S3 bucket
+  var s3 = S3.getInstance("<AWS_ACCESS_KEY>", "<AWS_SECRET_KEY>");
```

Comment:
Can you model this IAM user / role in CDK?

Reply:
Just added IAM user as "cdk-log-s3-user"